### PR TITLE
Application Connector Docs: Added a chapter describing authentication methods

### PR DESCRIPTION
**Pull request by akgalwas**
**Description**

Changes proposed in this pull request:

- Added information on authentication methods and OData to the overview (#3440 )
- Changed description of using external token (#2614)

**Related issue(s)**

### CODE REVIEW

File: docs/application-connector/docs/01-02-application-connector-external-apis.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
+---
+title: APIs exposed by the external solution
+
+---
+
+## Security
```
Comment (klaudiagrz) [inline docs/application-connector/docs/01-02-application-connector-external-apis.md]: I'd remove this heading

File: docs/application-connector/docs/01-02-application-connector-external-apis.md [hunk 2]
```diff
@@ -0,0 +1,24 @@
+---
+title: APIs exposed by the external solution
+
```
Comment (klaudiagrz) [inline docs/application-connector/docs/01-02-application-connector-external-apis.md]: Add this metadata entry to this and the 01-01 file:
```
type: Overview
```

The other option is to merge this document to the 01-01 as the separate ``## APIs exposed by the external solution`` heading.

File: docs/application-connector/docs/01-02-application-connector-external-apis.md [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+---
+
+## Security
+
+The Application Connector allows to register secured REST APIs exposed by the connected external solution. The Application Connector supports a variety of authentication methods to ensure smooth integration with a wide range of APIs.
```
[the remark this answers is not in the document]
Comment (akgalwas) [reply docs/application-connector/docs/01-02-application-connector-external-apis.md]: Done.

File: docs/application-connector/docs/01-02-application-connector-external-apis.md [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+
+The Application Connector allows to register secured REST APIs exposed by the connected external solution. The Application Connector supports a variety of authentication methods to ensure smooth integration with a wide range of APIs. 
+
+Your can register an API secured with one of the following:
+
+- Basic Authentication
+- OAuth
+- Client Certificates
+
+> **NOTE:** You can register non-secured API for testing purposes, however, it is not recommended in production environment.
```
[the remark this answers is not in the document]
Comment (akgalwas) [reply docs/application-connector/docs/01-02-application-connector-external-apis.md]: Done

File: docs/application-connector/docs/01-02-application-connector-external-apis.md [hunk 5]
```diff
@@ -0,0 +1,24 @@
+---
+title: APIs exposed by the external solution
+
+---
+
+## Security
+
+The Application Connector allows to register secured REST APIs exposed by the connected external solution. The Application Connector supports a variety of authentication methods to ensure smooth integration with a wide range of APIs. 
+
+Your can register an API secured with one of the following:
```
[the remark this answers is not in the document]
Comment (akgalwas) [reply docs/application-connector/docs/01-02-application-connector-external-apis.md]: Done

File: docs/application-connector/docs/01-02-application-connector-external-apis.md [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+
+- Basic Authentication
+- OAuth
+- Client Certificates
+
+> **NOTE:** You can register non-secured API for testing purposes, however, it is not recommended in production environment.
+
+In addition to authentication methods the Application Connector supports Cross-Site Request Forgery Tokens.
```
[the remark this answers is not in the document]
Comment (akgalwas) [reply docs/application-connector/docs/01-02-application-connector-external-apis.md]: Done

File: docs/application-connector/docs/01-02-application-connector-external-apis.md [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+
+## Types of APIs supported
+
+You can register any API adhering to the REST principles and available over HTTP protocol. The Application Connector also allows to register APIs implemented with OData technology.
```
[the remark this answers is not in the document]
Comment (akgalwas) [reply docs/application-connector/docs/01-02-application-connector-external-apis.md]: Done

File: docs/application-connector/docs/01-02-application-connector-external-apis.md [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+
+> **NOTE:** You can register non-secured API for testing purposes, however, it is not recommended in production environment.
+
+In addition to authentication methods the Application Connector supports Cross-Site Request Forgery Tokens.
+
+## Types of APIs supported
+
+You can register any API adhering to the REST principles and available over HTTP protocol. The Application Connector also allows to register APIs implemented with OData technology. 
+
+You can provide specifications documenting your APIs. The Application Connector supports [OpenAPI](https://www.openapis.org/) and [OData](https://www.odata.org/documentation) specification formats.
```
[the remark this answers is not in the document]
Comment (akgalwas) [reply docs/application-connector/docs/01-02-application-connector-external-apis.md]: Done

File: docs/application-connector/docs/01-02-application-connector-external-apis.md (hunk 1 above)
Comment (akgalwas) [reply docs/application-connector/docs/01-02-application-connector-external-apis.md]: Done

File: docs/application-connector/docs/01-02-application-connector-external-apis.md (hunk 2 above)
Comment (akgalwas) [reply docs/application-connector/docs/01-02-application-connector-external-apis.md]: Done

File: docs/application-connector/docs/03-05-pass-access-token-in-request-header.md [hunk 9]
```diff
@@ -7,7 +7,7 @@ The Application Connector supports passing the access token directly in the requ
 
 ## Passing the access token
 
-If the user is already authenticated to the service deployed on Kyma, the access token can be passed in a custom `Access-token` header. If the Application Connector detects that the custom header is present, instead of obtaining a new token, it passes the received one as a `Bearer` token in the `Authorization` header.
+If the user is already authenticated to the target API, the access token can be passed in a custom `Access-token` header. The value of  the header is of the form `Bearer {token}`. If the Application Connector detects that the custom header is present, instead of performing authentication steps, it removes `Access-token` header, and passes the received value in the `Authorization` header.
```
[the remark this answers is not in the document]
Comment (akgalwas) [reply docs/application-connector/docs/03-05-pass-access-token-in-request-header.md]: Done

File: docs/application-connector/docs/01-02-application-connector-external-apis.md [hunk 10]
```diff
@@ -0,0 +1,20 @@
+---
+title: APIs exposed by the external solution
```
Comment (klaudiagrz) [inline docs/application-connector/docs/01-02-application-connector-external-apis.md]:
```suggestion
title: Supported APIs
```

Comment (akgalwas) [reply docs/application-connector/docs/01-02-application-connector-external-apis.md]: Done

**Pull request merged**